source/vibe/web/rest.d

```diff
@@ -969,12 +987,13 @@ private string genClientBody(alias Func)() {
 				static if (is(PT == Json))
 					url_prefix = q{urlEncode(id.toString())~"/"};
 				else
-					url_prefix = q{urlEncode(toRestString(serializeToJson(id)))~"/"};
+					url_prefix = q{urlEncode(toRestString(id))~"/"};
 			} else static if (anySatisfy!(mixin(GenCmp!("ClientFilter", i, ParamNames[i]).Name), paramAttr)) {
 				alias paramsArgList = Filter!(mixin(GenCmp!("ClientFilter", i, ParamNames[i]).Name), UDATuple!(WebParamAttribute, Func));
-				static if (paramsArgList[0].origin == WebParamAttribute.Origin.Header)
-					param_handling_str ~= format(q{headers__["%s"] = to!string(%s);}, paramsArgList[0].field, paramsArgList[0].identifier);
-				else static if (paramsArgList[0].origin == WebParamAttribute.Origin.Query)
+				static if (paramsArgList[0].origin == WebParamAttribute.Origin.Header) {
+					static if (isNullable!PT) param_handling_str ~= format(q{ if (!%s.isNull) }, paramsArgList[0].identifier);
+					param_handling_str ~= format(q{headers__["%s"] = toRestString(%s);}, paramsArgList[0].field, paramsArgList[0].identifier);
```

Comment:
That actually sounds like a good way to do it, although it would be a brekaing change as well, because query parameters are passed as serialized Json right now.